pkg/controllers/managementuser/clusterauthtoken/token.go

```diff
@@ -101,6 +131,26 @@ func (h *tokenHandler) Updated(token *managementv3.Token) (runtime.Object, error
 		return nil, err
 	}
 
+	forced := false
+	clusterAuthTokenSecret, err := h.clusterSecretLister.Get(h.namespace, common.ClusterAuthTokenSecretName(token.Name))
+	if err != nil {
```

Comment:
Should we restrict this to just to `NotFound` errors?

Reply:
Basically add the below, right?
```suggestion
	if err != nil {
	    if !errors.IsNotFound(err) {
	        return nil, err
	    }
	    // err is not-found, 
```

Hm ... So, currently, if the lister returns some error E which is not `not found` ... we create the hash, create the secret in memory, update the user attributes, and at last, because of the force will definitely try to update or create the secret ... and any error from that is returned ... which may be E or not. and of course assumes that nothing in the sequence errors out either.

Ok, I am convinced now that the restriction and bail-out should be made, given that any other unexpected error causes the same.

Reply:
And checking with `kube-api-auth` ... its equivalent code has the restriction in question!
See https://github.com/rancher/kube-api-auth/blob/6da2d1cb468bb95de0ab019263ebf9783768f32d/pkg/service/handlers/v1-kube-api-authn.go#L131